### ISSUE TITLE
Number cell with BigInt support

### ISSUE THREAD

**Issue by LukasMasuch** [glideapps/glide-data-grid#843]
The current version of the number cell only supports normal `Numbers` but not `BigInt` numbers (> 2^53). Two potential solutions would be to:

1. Enable the native number-cell to handle `BigInt` values in addition to `Number` values. 
2. Add a dedicate big-int cell to the `cells` package.

**Comment by jassmith** [comment]
I think option 1 is the way to go if react number format supports it. Have you a usecase for bigint data?

**Comment by LukasMasuch** [comment]
yeah, users have run into this a couple of times mostly related to columns that contain numerical IDs.

**Comment by jassmith** [comment]
It looks like to make this possible we would need to swap out the editor implementation. react-number-format does not support this.

**Comment by LukasMasuch** [comment]
I think `react-number-format` should be fine with handling larger numbers. E.g. `9007199254740991` is the highest supported value for Number without losing precision. In [this demo](https://codesandbox.io/p/sandbox/onvaluechange-demo-c5nl2f), if you go higher the `floatValue` will lose precision but the string value will still show the correct number:

https://github.com/glideapps/glide-data-grid/assets/2852129/5c33642e-e1d8-43d2-853a-9cebaadefabc

This means, that we would need to handle the value as a string for the `react-number-format` input field if the cell data is a bigint.

**Comment by jassmith** [comment]
Oh very nice. Okay Im on board with this.
